python-regression/IXI/merge-spent/index.js

```diff
@@ -0,0 +1,147 @@
+var System = java.lang.System;
+
+var spentAddressProvider = IOTA.spentAddressesProvider
+
+var iri = com.iota.iri;
+var Callable = iri.service.CallableRequest;
+var Response = iri.service.dto.IXIResponse;
+var ErrorResponse = iri.service.dto.ErrorResponse;
+
+var defaultSpentFileName = "spentAddresses.txt";
+
+// Log using logger of the ixi class
+var log = org.slf4j.LoggerFactory.getLogger(iri.IXI.class);
+
+function checkChecksumFor(file){
+    var digest = java.security.MessageDigest.getInstance("SHA-256");
+
+    var stream = new java.io.FileInputStream(file);
+    var targetReader = new java.io.BufferedReader(new java.io.InputStreamReader(stream));
+    var expectedChecksum;
+
+    try {
+        var firstLine = true;
+        while (line = targetReader.readLine()){
+            if (firstLine){
+                firstLine = false;
+            } else if (line.length() === 64){
+                // Checksum
+                expectedChecksum = line;
+            } else if (line.length() === 81){
+                // Hash
+                digest.update(line.getBytes());
+            }
+        }
+
+        targetReader.close();
+        stream.close();
+    } catch (error){
+        targetReader.close();
+        stream.close();
+
+        log.error("Caught error during read: {}", error);
+        return false;
+    }
+
+    if (!expectedChecksum){
+        log.error("{} did not have a checksum", file.getName());
+        return false;
+    }
+
+    return expectedChecksum !== checksumFromBytes(digest.digest());
+}
+
+function updateSpentAddressesWith(file){
+    if (!checkChecksumFor(file)){
+        return {
+            "error": "Checksum mismatch for file " + file.getName()
+        }
+    }
+
+    var stream = new java.io.FileInputStream(file);
+    var targetReader = new java.io.BufferedReader(new java.io.InputStreamReader(stream));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['java' is not defined. (no-undef)](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=3955655)